umh-core/pkg/communicator/actions/deploy-dataflowcomponent_test.go

```diff
@@ -0,0 +1,420 @@
+									"data": "type: mapping\nprocs: []",
+								},
+							},
+						},
+					},
+				},
+			}
+
+			err := action.Parse(payload)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Reset tracking for this test
+			mockConfig.ResetCalls()
+
+			// Execute the action
+			result, metadata, err := action.Execute()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).To(BeNil())
+			Expect(metadata).To(BeNil())
+
+			// We should have messages in the channel (Confirmed, then Success)
+			var messages []*models.UMHMessage
+			for i := 0; i < 2; i++ {
+				select {
+				case msg := <-outboundChannel:
+					messages = append(messages, msg)
+				case <-time.After(100 * time.Millisecond):
+					Fail("Timed out waiting for message")
+				}
+			}
+			Expect(messages).To(HaveLen(2))
+
+			// Verify AtomicAddDataflowcomponent was called
+			Expect(mockConfig.AddDataflowcomponentCalled).To(BeTrue())
+
+			Expect(mockConfig.Config.DataFlow).To(HaveLen(1))
+			Expect(mockConfig.Config.DataFlow[0].Name).To(Equal("test-component"))
+			Expect(mockConfig.Config.DataFlow[0].DesiredFSMState).To(Equal("running"))
+		})
+
+		It("should handle AtomicAddDataflowcomponent failure", func() {
+			// Set up mock to fail on AtomicAddDataflowcomponent
+			mockConfig.WithAddDataflowcomponentError(errors.New("mock add dataflow component failure"))
+
+			// Parse with valid payload
+			payload := map[string]interface{}{
+				"name": "test-component",
+				"meta": map[string]interface{}{
+					"type": "custom",
+				},
+				"ignoreHealthCheck": false,
+				"payload": map[string]interface{}{
+					"customDataFlowComponent": map[string]interface{}{
+						"inputs": map[string]interface{}{
+							"type": "yaml",
+							"data": "input: something\nformat: json",
+						},
+						"outputs": map[string]interface{}{
+							"type": "yaml",
+							"data": "output: something\nformat: json",
+						},
+						"pipeline": map[string]interface{}{
+							"processors": map[string]interface{}{
+								"proc1": map[string]interface{}{
+									"type": "yaml",
+									"data": "type: mapping\nprocs: []",
+								},
+							},
+						},
+					},
+				},
+			}
+
+			err := action.Parse(payload)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Execute the action - should fail
+			result, metadata, err := action.Execute()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("Failed to add dataflowcomponent"))
+			Expect(result).To(BeNil())
+			Expect(metadata).To(BeNil())
+
+			// We should have failure messages in the channel
+			var messages []*models.UMHMessage
+			for i := 0; i < 2; i++ {
+				select {
+				case msg := <-outboundChannel:
+					messages = append(messages, msg)
+				case <-time.After(100 * time.Millisecond):
+					Fail("Timed out waiting for message")
+				}
+			}
+			Expect(messages).To(HaveLen(2))
+
+			// For failure tests, we don't check exact content since it may be encoded
+			// We check that we received the expected number of messages,
```

Comment:
the expected number of messages is the same in the success as in the failure case, need to actually look into it to check for failure